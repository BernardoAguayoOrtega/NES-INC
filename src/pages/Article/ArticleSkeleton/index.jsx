// import react
import React from 'react';
//import react-loading-skeleton
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';

const ArticleSkeleton = () => {
  return (
    <SkeletonTheme color="#36404e" highlightColor="#2E2E32">
      <article className="post">
        <section className="post__content">
          <div className="post__content--headline">
            <h1>
              <Skeleton height={100}/>
            </h1>
          </div>
          <div className="post__content--image">
            <Skeleton height={400}/>
          </div>
          <div className="post__content--text">
            <p>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
            </p>
            <br/>
            <p>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
            </p>
            <br/>
            <p>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
              <Skeleton/>
            </p>
          </div>
        </section>
      </article>
    </SkeletonTheme>
  )
};

export default ArticleSkeleton;
