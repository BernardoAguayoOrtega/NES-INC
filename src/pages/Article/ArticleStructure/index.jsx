// import react
import React, { useContext, useEffect, useState } from 'react';
// use params hooks
import { useParams } from 'react-router-dom'
// import context
import { Context } from '../../../utils/Context/index.jsx';
// import disqus
import { DiscussionEmbed } from 'disqus-react';

const ArticleStructure = () => {
  // use state
  const [new$, setNew$] = useState()
  // use context
  const { news } = useContext(Context)
  // use ref hooks
  const { name } = useParams()
  // use effect
  useEffect(() => {
    setNew$(news?.data?.find(item => item?.author === name && item))
  }, [])

  return (
    <article className="post">
      <section className="post__content">
        <div className="post__content--headline">
          <h1>{new$?.title}</h1>
        </div>
        <img src={`http://${new$?.images[0]}` || 'https://images.unsplash.com/photo-1441974231531-c6227db76b6e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1051&q=80CdYg35qC4hY8ByncPpj_6MQ&usqp=CAU '} className="post__content--image"/>
        <div className="post__content--text">
          {new$?.body?.map((paragraph, i) => <p key={i}>{paragraph}</p>)}
        </div>
      </section>
      <DiscussionEmbed
        shortname="news-inc"
        config={
          {
            url: `http://localhost:3000/#/article/${new$?.author}`,
            identifier: new$?._id,
            title: new$?.title,
            language: 'en'
          }
        }
      />
    </article>
  )
};

export default ArticleStructure;
