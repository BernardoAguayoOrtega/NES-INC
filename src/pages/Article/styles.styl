@import '../../stylus/variables.styl'

.post
  margin: 0 auto
  max-width: 1200px
  &__content
    margin: 0 auto
    padding-top: 10%
    margin-bottom: 2rem
    max-width: 800px
    &--headline
      h1
        text-align: center
        font-family: Font1
        font-size: 8rem
        letter-spacing: .5rem
        line-height: 5rem
      margin-bottom: 1rem
    &--image
      margin-top: 1.8rem
      object-fit: cover
      width: 100%
      height: 400px
      border-radius: 2px
    &--text
      margin-top: 2rem
      p
        color: text;
        text-align: justify
        line-height: 2rem
        font-family: Font2
        font-size: 1.28rem
    &--quote
      display: flex
      justify-content: center
      margin-top: 2rem
      h3
        color: #36404E
        text-align: justify
        line-height: 1.5rem
        font-family: Font2
        font-size: 2rem
      a
        text-decoration: none
        &:hover
          text-decoration: underline #36404E
  &__comments
      margin: 2rem auto
      max-width: 830px

@media screen and (max-width: 480px)
  .post
    &__content
      &--headline
        h1
          font-size: 2rem
          line-height: 2rem

@media screen and (min-width: 480px)
  .post
    &__content
      &--headline
        h1
          font-size: 3rem
          line-height: 3rem

@media screen and (max-width: 520px)
  .post
    &__content
      &--text
        p
          font-size: 1rem
      &--quote
        h3
          font-size: 1.5rem

@media screen and (min-width: 580px)
  .post
    &__content
      &--headline
        h1
          font-size: 4rem
          line-height: 3.5rem

@media screen and (min-width: 761px)
  .post
    &__content
      &--headline
        h1
          font-size: 5rem
          line-height: 4.5rem

@media screen and (max-width: 1200px)
  .post
    width: 90%
    margin: 0 auto
    &__content
      width: 100%
      &--image
        width: 100%

@media screen and (min-width: 1200px)
  .post
    width: 1200px
