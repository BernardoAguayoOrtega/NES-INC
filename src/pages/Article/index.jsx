// import react
import React, { useContext, useEffect, useState } from 'react';
// use params hooks
import { useParams } from 'react-router-dom'
// import context
import { Context } from '../../utils/Context/index.jsx';
// import disqus
import { DiscussionEmbed } from 'disqus-react';
// import styles
import './styles.styl';

// create and export Home page
const Article = () => {
  // use state
  const [new$, setNew$] = useState()
  // use context
  const { news } = useContext(Context)
  // use ref hooks
  const { name } = useParams()
  // use effect
  useEffect(() => {
    setNew$(news?.data?.find(item => item?.author === name && item))
  }, [])

  return (
    <article className="post">
      <section className="post__content">
        <div className="post__content--headline">
          <h1>{new$?.title}</h1>
        </div>
        <img src={new$?.image || 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcR3OACZKecLGqNCdYg35qC4hY8ByncPpj_6MQ&usqp=CAU '} className="post__content--image"/>
        <div className="post__content--text">
          {new$?.body?.map((paragraph, i) => <p key={i}>{paragraph}</p>)}
        </div>
      </section>
      <DiscussionEmbed
        shortname="article"
        config={
          {
            url: '',
            identifier: '',
            title: '',
            language: 'en' //e.g. for Traditional Chinese (Taiwan)
          }
        }
      />
    </article>
  )
};

export default Article;
