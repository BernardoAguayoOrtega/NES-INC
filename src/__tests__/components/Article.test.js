// import react
import React from 'react';
// import shallow
import { mount } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import context
import { ContextProvider } from '../../utils/Context/index.jsx';
// import app component
import Article from '../../pages/Article/index.jsx';

describe('<Article/>', () => {
  const wrapper = mount(
    <ContextProvider>
      <Article />
    </ContextProvider>,
  );

  it('is rendering the Article?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is rendering the skeleton?', () => {
    expect(wrapper.find('ArticleSkeleton').exists()).toBe(true);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
