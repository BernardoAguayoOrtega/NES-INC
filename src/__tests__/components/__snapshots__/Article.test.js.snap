// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article ...props/> matches the snap shot 1`] = `
<ContextProvider
  $news={
    Object {
      "image": "random",
      "title": "random",
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Article>
        <article
          className="post"
        >
          <section
            className="post__content"
          >
            <div
              className="post__content--headline"
            >
              <h1 />
            </div>
            <img
              className="post__content--image"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcR3OACZKecLGqNCdYg35qC4hY8ByncPpj_6MQ&usqp=CAU "
            />
            <div
              className="post__content--text"
            />
          </section>
          <DiscussionEmbed
            config={
              Object {
                "identifier": "",
                "language": "en",
                "title": "",
                "url": "",
              }
            }
            shortname="article"
          >
            <div
              config={
                Object {
                  "identifier": "",
                  "language": "en",
                  "title": "",
                  "url": "",
                }
              }
              id="disqus_thread"
              shortname="article"
            />
          </DiscussionEmbed>
        </article>
      </Article>
    </Router>
  </HashRouter>
</ContextProvider>
`;
